### PR TITLE
Track player events

### PR DESCRIPTION
**Pull request by ashiagr**
| 📘 Project: #261  |
|:---:|

Corresponding iOS PR: https://github.com/Automattic/pocket-casts-ios/pull/289

This adds the basic events to the full screen player view: 

- `player_shown`: When the full screen player is displayed to the user
- `player_dismissed`: When the user dismisses the full screen player
- `player_tab_selected`: When one of the tabs at the top of the player view is selected
- `player_show_notes_link_tapped`: When the user taps a link in the player show notes
- `player_chapter_selected`: When the user selects a chapter from the chapter list tab
- `player_previous_chapter_tapped`: When the user taps the chapter back button
- `player_next_chapter_tapped`: When the user taps the chapter forward button

## To test

1. Play an episode if you don't have one playing already
2. Tap the mini player
3. ✅ Verify you see `🔵 Tracked: player_shown` in console
4. Dismiss the player either by tapping the down arrow or swiping it away
5. ✅ Verify you see `🔵 Tracked: player_dismissed`
6. Open the player again
7. Tap the Notes tab at the top
8. ✅ `🔵 Tracked: player_tab_selected ["tab": "show_notes"]`
9. Tap a link in the show notes
10. ✅ `🔵 Tracked: player_show_notes_link_tapped ["episode_uuid": "{UUID}"]` - Where {UUID} is the uuid of the episode that is playing
11. Play an episode that has chapters
- An example one is: https://pca.st/upgrade (E414)
12. Tap on the Chapters tab
13. ✅ `🔵 Tracked: player_tab_selected ["tab": "chapters"]`
14. Select one of the chapters in the list
15. ✅ `🔵 Tracked: player_chapter_selected`
16. Go back to the Now Playing tab
17. ✅ `🔵 Tracked: player_tab_selected ["tab": "now_playing"]`
18. Tap the next chapter button
19. ✅ `🔵 Tracked: player_next_chapter_tapped`
20. Tap the previous chapter button
21. ✅ `🔵 Tracked: player_previous_chapter_tapped`

# Checklist
N/A

- [ ] Should this change be included in the release notes? If yes, please add a line in CHANGELOG.md
- [ ] Have you tested in landscape?
- [ ] Have you tested accessibility with TalkBack?
- [ ] Have you tested in different themes?
- [ ] Does the change work with a large display font?
- [ ] Are all the strings localized?
- [ ] Could you have written any new tests?
- [ ] Did you include Compose previews with any components?

### CODE REVIEW

File: modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/view/PlayerContainerFragment.kt [hunk 1]
```diff
@@ -120,8 +120,14 @@ class PlayerContainerFragment : BaseFragment(), HasBackstack {
...
-                    1 -> FirebaseAnalyticsTracker.openedPlayerNotes()
-                    2 -> FirebaseAnalyticsTracker.openedPlayerChapters()
+                    1 -> {
+                        analyticsTracker.track(AnalyticsEvent.PLAYER_TAB_SELECTED, mapOf(TAB_KEY to SHOW_NOTES))
+                        FirebaseAnalyticsTracker.openedPlayerNotes()
+                    }
+                    2 -> {
+                        analyticsTracker.track(AnalyticsEvent.PLAYER_TAB_SELECTED, mapOf(TAB_KEY to CHAPTERS))
+                        FirebaseAnalyticsTracker.openedPlayerChapters()
+                    }
```
[the remark this answers is not in the document]
Comment (mchowning) [reply modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/view/PlayerContainerFragment.kt:130]: Might be worth also adding an else clause here and log a warning. That might help us catch if we miss a tab at some point in the future. I don't feel strongly.

**Review by mchowning — COMMENTED** [review]
This looks good. Only issue I saw was us sending the tab tapped event when the Now Playing tab is tapped.

File: modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/view/PlayerContainerFragment.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (ashiagr) [reply modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/view/PlayerContainerFragment.kt:134]: Both are valid suggestions. I added in 1753b8e.

[the remark this answers is not in the document]
Comment (mchowning) [reply modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/view/PlayerContainerFragment.kt:134]: It's now detecting the player tab being selected well, 🙂  but I'm also seeing that event getting sent if I kill the app and then restart it. 🤔 

https://user-images.githubusercontent.com/4656348/194919273-375c1647-cb9a-4099-a79d-77c33056dfd1.mov

[the remark this answers is not in the document]
Comment (ashiagr) [reply modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/view/PlayerContainerFragment.kt:136]: The tab is auto-selected when the view pager is created. I'm preventing this tracking in d1fdcac.

**Review by mchowning — APPROVED** [review]
Looks good! :shipit:

**Pull request merged**